lib/localDev.ts

```diff
@@ -1,66 +1,77 @@
   confirmUseExistingDeveloperTestAccountPrompt,
-} = require('./prompts/projectDevTargetAccountPrompt');
-const { confirmPrompt } = require('./prompts/promptUtils');
-const {
-  validateSandboxUsageLimits,
-  getAvailableSyncTypes,
-} = require('./sandboxes');
-const { syncSandbox } = require('./sandboxSync');
-const {
-  validateDevTestAccountUsageLimits,
-} = require('./developerTestAccounts');
-const { uiCommandReference, uiLine, uiAccountDescription } = require('./ui');
-const SpinniesManager = require('./ui/SpinniesManager');
-const { i18n } = require('./lang');
-const { EXIT_CODES } = require('./enums/exitCodes');
-const { trackCommandMetadataUsage } = require('./usageTracking');
-const {
-  isAppDeveloperAccount,
-  isDeveloperTestAccount,
-} = require('./accountTypes');
-const { handleProjectUpload } = require('./projects/upload');
-const { pollProjectBuildAndDeploy } = require('./projects/buildAndDeploy');
-const {
+} from './prompts/projectDevTargetAccountPrompt';
+import { confirmPrompt } from './prompts/promptUtils';
+import { validateSandboxUsageLimits, getAvailableSyncTypes } from './sandboxes';
+import { syncSandbox } from './sandboxSync';
+import { validateDevTestAccountUsageLimits } from './developerTestAccounts';
+import { uiCommandReference, uiLine, uiAccountDescription } from './ui';
+import SpinniesManager from './ui/SpinniesManager';
+import { i18n } from './lang';
+import { EXIT_CODES } from './enums/exitCodes';
+import { trackCommandMetadataUsage } from './usageTracking';
+import { isAppDeveloperAccount, isDeveloperTestAccount } from './accountTypes';
+import { handleProjectUpload } from './projects/upload';
+import { pollProjectBuildAndDeploy } from './projects/buildAndDeploy';
+import {
   PROJECT_ERROR_TYPES,
   PROJECT_BUILD_TEXT,
   PROJECT_DEPLOY_TEXT,
-} = require('./constants');
-const { logError, ApiErrorContext } = require('./errorHandlers/index');
-const {
-  PERSONAL_ACCESS_KEY_AUTH_METHOD,
-} = require('@hubspot/local-dev-lib/constants/auth');
-const {
+} from './constants';
+import { logError, ApiErrorContext } from './errorHandlers/index';
+import {
   buildSandbox,
   buildDeveloperTestAccount,
   saveAccountToConfig,
-} = require('./buildAccount');
-const { hubspotAccountNamePrompt } = require('./prompts/accountNamePrompt');
+} from './buildAccount';
+import { hubspotAccountNamePrompt } from './prompts/accountNamePrompt';
+import {
+  ProjectConfig,
+  ProjectPollResult,
+  ProjectSubtask,
+} from '../types/Projects';
+import { FileResult } from 'tmp';
+import { Build } from '@hubspot/local-dev-lib/types/Build';
 
 const i18nKey = 'lib.localDev';
 
 // If the user passed in the --account flag, confirm they want to use that account as
 // their target account, otherwise exit
-const confirmDefaultAccountIsTarget = async accountConfig => {
+export async function confirmDefaultAccountIsTarget(
+  accountConfig: CLIAccount
+): Promise<void> {
+  if (!accountConfig.name || !accountConfig.accountType) {
```

Comment:
I'm a little wary of adding new logs in a TS port since it could subtly change the functionality. Would it be possible to break the code additions out into a standalone PR to merge and test before this one?

Reply:
Any new logs added here are more or less impossible to actually reach. I added them to satisfy TS (throwing errors makes typing a lot easier than having to detail with returning `undefined`). I guess theoretically this would trigger if you went in and manually removed these fields from the config but I have to imagine this would be better than whatever the behavior was previously. If you still think its worth separating out I can do that though